### PR TITLE
Added focus on slash key press and unfocus on esc

### PR DESCRIPTION
**Pull request by tanmaylaud**
Closes https://github.com/vercel/next.js/issues/14944

This PR fixes issue [14944](https://github.com/vercel/next.js/issues/14944) in Nextjs docs.

I have add listener for both "/" and "Escape" to focus and blur respectively.

### CODE REVIEW

**Comment by tanmaylaud** [comment]
@lfades Could you please review this PR? Thanks !

**Comment by tanmaylaud** [comment]
> Hi @tanmaylaud thank you for the PR!
> 
> Please make sure to allow the key (`/`) to be used in other inputs as requested in the issue. For example, right now you can't type `/` inside the search input.
> 
> You can fix this by ignoring any input that passes this check:
> 
> ```js
> el.contentEditable === 'true' || el.tagName == 'INPUT' || el.tagName == 'TEXTAREA' || el.tagName == 'SELECT'
> ```

@lfades  Thanks for the observation. Completely missed that part. I have added it now. Please check.

File: components/search/auto-complete.js [hunk 1]
```diff
@@ -59,6 +60,32 @@ function AutoComplete({
...
+      )
+        return;
+      // focus input on pressing the slash key
+      textInput.current.focus();
+      event.preventDefault();
+    } else if (event.key === 'Escape') {
+      document.activeElement.blur();
+      event.preventDefault();
+    }
+  }, []);
```
Comment (lfades) [inline components/search/auto-complete.js:81]:
```suggestion
  }, [textInput.current]);
```

File: components/search/auto-complete.js [hunk 2]
```diff
@@ -59,6 +60,32 @@ function AutoComplete({
...
+
+  useEffect(() => {
+    document.addEventListener('keydown', focusSearch, false);
+    return () => {
+      document.removeEventListener('keydown', focusSearch, false);
+    };
+  }, []);
```
Comment (lfades) [inline components/search/auto-complete.js:88]:
```suggestion
  }, [focusSearch]);
```

**Pull request merged**